### PR TITLE
delay getting the home dir of a user untill needed

### PR DESCRIPTION
**Pull request by icewind1991**
for cache-only operations we don't actually need to know the home directory

Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Review by CarlSchwan — APPROVED** [review]

File: lib/private/Files/Storage/Home.php [hunk 1]
```diff
@@ -49,12 +49,24 @@ class Home extends Local implements \OCP\Files\IHomeStorage {
...
-		parent::__construct(['datadir' => $datadir]);
+		// use a placeholder datadir until we actually need to caluclate a source path
+		//
+		// this allows using this storage with a LazyUser without having to get the real user
+		// as long as only cache operations are done
+		parent::__construct(['datadir' => '/tmp/empty/placeholder/']);
 	}
 
+	public function getSourcePath($path) {
+		if ($this->datadir == '/tmp/empty/placeholder/') {
```
Comment (PVince81) [inline lib/private/Files/Storage/Home.php:62]: use a constant ?

should we use a value even more cryptic ?

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
conflict

**Comment by szaimen** [comment]
@icewind1991 please fix the test

**Pull request closed without merging**